lib/cursor.js

```diff
@@ -4,21 +4,35 @@ const common = require('metarhia-common');
 
 const core = require('./core');
 
+const defaultOptions = {
+  jsql: [],
+  parents: [],
+  timeout: 5000,
+  category: null,
+  schema: null,
+  provider: null,
+};
+
 class Cursor {
-  constructor() {
-    this.parents = [];
-    this.provider = null;
+  constructor(options) {
     this.dataset = [];
     this.children = [];
-    this.jsql = [];
-    this.schema = null;
+    Object.keys(defaultOptions).forEach(key => {
```

Comment:
I don't like such implicit style, I used it everywhere a few years ago but stopped doing that for the following reasons:
- This may affect performance and may block optimization
- Explicit is better than the implicit, not everywhere, but in this case it reduces readability
- I don't think we need options in this certain case

Reply:
I'm not sure how it affects performance and readability as you avoid boilerplate code of the form:
```
this.opt1 = options || options[opt1] === undefined ? this.defaultOptions[opt1] : this.options[opt1];
this.opt2 = options || options[opt2] === undefined ? this.defaultOptions[opt2] : this.options[opt2];
this.opt3 = options || options[opt3] === undefined ? this.defaultOptions[opt3] : this.options[opt3];
...
```
Have all default options packed in a single object instead of around the code.

As for the performance, this only happens during the creation of a Cursor though if this is one of the main operations for the Cursor then indeed this may have a slight performance implications but without proper benchmarks I'd say that we may neglect it for now.

> I don't think we need options in this certain case

Well you manually assigned those variables [here](https://github.com/metarhia/globalstorage/pull/215/files#diff-14b597bc6604e1b7ec69ae786619e979L106) and [here](https://github.com/metarhia/globalstorage/pull/215/files#diff-7677206856daeffa7bde6930bf32de9aL152) so I concluded that it indeed needs options.